docs/tutorials/notes-mvvm/includes/viewmodel-note.md

```diff
@@ -0,0 +1,130 @@
+
+01. Finally, add these two helper methods to the class:
+
+    :::code language="csharp" source="../snippets/viewmodel-shared/csharp/Notes/ViewModels/NoteViewModel.cs" id="helpers":::
+
+    The `Reload` method is a helper method that refreshes the backing model object, reloading it from device storage
+
+    The `RefreshProperties` method is another helper method to ensure that any subscribers bound to this object are notified that the `Text` and `Date` properties have changed. Since the underlying model (the `_note` field) is changed when the note is loaded during navigation, the `Text` and `Date` properties aren't actually set to new values. Since these properties aren't directly set, any bindings attached to those properties wouldn't be notified because `OnPropertyChanged` isn't called for each property. `RefreshProperties` ensures bindings to these properties are refreshed.
+
+The code for the class should look like the following snippet:
+
+:::code language="csharp" source="../snippets/viewmodel-shared/csharp/Notes/ViewModels/NoteViewModel.cs" id="full":::
+
+## Note view
+
+Now that the viewmodel has been created, update the **Note view**. In the _Views\\NotePage.xaml_ file, apply the following changes:
+
+- Add the `xmlns:viewModels` XML namespace that targets the `Notes.ViewModels` .NET namespace.
+- Add a `BindingContext` to the page.
+- Remove the delete and save button `Clicked` event handlers and replace them with commands.
+
+Update the **Note view**:
+
+01. In the **Solution Explorer** pane of Visual Studio, double-click on **Views\\NotePage.xaml** to open the XAML editor.
+01. Paste in the following code:
+
+:::code language="xaml" source="../snippets/viewmodel-shared/csharp/Notes/Views/NotePage.xaml" highlight="4,7-9,18,22":::
+
+Previously, this view didn't declare a binding context, as it was supplied by the code-behind of the page itself. Setting the binding context directly in the XAML provides two things:
+
+- At run-time, when the page is navigated to, it displays a blank note. This is because the parameterless constructor for the binding context, the viewmodel, is invoked. If you remember correctly, the parameterless constructor for the **Note viewmodel** creates a blank note.
+
+- The intellisense in the XAML editor shows the available properties as soon as you start typing `{Binding` syntax. The syntax is also validated and alerts you of an invalid value. Try changing the binding syntax for the `SaveCommand` to `Save123Command`. If you hover the mouse cursor over the text, you'll notice that a tooltip is displayed informing you that **Save123Command** isn't found. This notification isn't considered an error because bindings are dynamic, it's really a small warning that may help you notice when you typed the wrong property.
```

Comment:
Not really relevant here but you can turn on compiled bindings. Part of that is that you'll get a build error for a binding that doesn't resolve.